learnEcl/2700-function.md

```diff
@@ -2,3 +2,147 @@
 title: Function
 slug: function
 ---
+
+# FUNCTION
+
+A Function is a set of statements that take inputs, does some specific computation and produces an output or return a result. Result could be a value or a dataset.
+
+Notes
+
+* For function to be called/used from outside, EXPORT is required
+* Function name should match the file name. if not "Error: Definition must contain EXPORT or SHARED " is generated
```

Comment:
A non-BWR file must have an EXPORTed attribute that matches the filename.  That attribute can be anything, not just a function.  Most commonly it is a MODULE to wrap other exported functions.  In all cases, the EXPORTed attribute must be the *last* exported attribute in the file.  It is perfectly legal to have other (local or SHARED) attributes preceding the final EXPORTed attribute.